extensions/data-plane/data-plane-aws-s3/src/main/java/org/eclipse/edc/connector/dataplane/aws/s3/S3DataSinkFactory.java

```diff
@@ -128,4 +132,14 @@ private S3ClientRequest createS3ClientRequest(DataAddress address) {
             return S3ClientRequest.from(region, endpointOverride);
         }
     }
+    
+    private SecretToken deserializeSecretToken(String secret) {
+        var typeReference = new TypeReference<HashMap<String, Object>>() {};
+        var map = typeManager.readValue(secret, typeReference);
```

Comment:
This was the old way of deserializing things, lately we stopped going through the `TypeManager` but we're using the `ObjectMapper` instance directly, by doing that it will enable to use the `readTree` method to read the tree (that can tell if the "sessionToken" exists) and then with the `treeToValue` the tree can be converted to the actual type.